tests/src/Functional/OgSelectionWidgetOptionsTest.php

```diff
@@ -0,0 +1,186 @@
+
+  /**
+   * Administrator groups user.
+   *
+   * @var \Drupal\user\Entity\User
+   */
+  protected $groupAdministratorUser;
+
+  /**
+   * A non-member user.
+   *
+   * @var \Drupal\user\Entity\User
+   */
+  protected $nonMemberUser;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    parent::setUp();
+
+    // Create group node types.
+    $this->createContentType(['type' => 'group_type1']);
+    $this->createContentType(['type' => 'group_type2']);
+
+    Og::addGroup('node', 'group_type1');
+    Og::addGroup('node', 'group_type2');
+
+    NodeType::create(['type' => 'group_content'])->save();
+
+    // Use a select list widget for the audience field, so it's easier to get
+    // all the values.
+    $settings = [
+      'form_display' => [
+        'type' => 'options_select',
+      ],
+    ];
+    Og::createField(OgGroupAudienceHelper::DEFAULT_FIELD, 'node', 'group_content', $settings);
+
+    // Create users.
+    $this->groupMemberUser = $this->drupalCreateUser();
+    $this->groupOwnerUser = $this->drupalCreateUser();
+    $this->groupAdministratorUser = $this->drupalCreateUser(['administer organic groups']);
+    $this->nonMemberUser = $this->drupalCreateUser();
+
+    // Create groups.
+    $this->group1 = Node::create([
+      'type' => 'group_type1',
+      'title' => 'group1',
+      'uid' => $this->groupOwnerUser->id(),
+    ]);
+    $this->group1->save();
+
+    $this->group2 = Node::create([
+      'type' => 'group_type2',
+      'title' => 'group2',
+      'uid' => $this->groupOwnerUser->id(),
+    ]);
+    $this->group2->save();
+
+    $this->unpublishedGroup = Node::create([
+      'type' => 'group_type2',
+      'title' => 'unpublished group',
+      'uid' => $this->groupOwnerUser->id(),
+      'status' => NodeInterface::NOT_PUBLISHED,
+    ]);
+    $this->unpublishedGroup->save();
+
+    // Add member to group.
+    Og::createMembership($this->group1, $this->groupMemberUser)->save();
+    Og::createMembership($this->group2, $this->groupMemberUser)->save();
+  }
+
+  /**
+   * Tests the group audience widgets shows correct values.
+   */
+  public function testNonRequiredAudienceField() {
+    // Non member user.
+    $this->drupalLogin($this->nonMemberUser);
+    $this->drupalGet('node/add/group_content');
+    $this->assertSession()->statusCodeEquals(403);
+
+    // Group member without create permissions.
+    $this->drupalLogin($this->groupMemberUser);
+    $this->drupalGet('node/add/group_content');
+    $this->assertSession()->statusCodeEquals(403);
+
+    // Grant create permission for the first group.
+    $role = OgRole::getRole($this->group1->getEntityTypeId(), $this->group1->bundle(), OgRoleInterface::AUTHENTICATED);
+    $role
+      ->grantPermission('create group_content content')
+      ->save();
+
+    $this->drupalGet('node/add/group_content');
+    // The user can't post group content site wide so the audience is required
+    // field.
+    // todo: is that correct?
```

Comment:
Minor coding standards adjustment.
```suggestion
    // todo Is that correct?
```